File: Problem_7/problem_7.fs
open System

let list_of_dividers num =
    [2 .. int (floor (sqrt (float num))) + 1]
    |> Seq.filter (fun x -> num % x = 0)

let is_prime num =
    (Seq.length (list_of_dividers num)) = 0

let rec problem_7 current primes counter =
    match primes, counter with
        | x, y when x >= 10002 -> current
        | x, y when x< 10002 && (is_prime counter) -> problem_7 counter (primes + 1) (counter + 1)
        | _, _ -> problem_7 current primes (counter + 1)

[<EntryPoint>]
let main(args : string[]) =
    printfn "result = %d" (problem_7 0 0 0)
    0

File: Problem_07/problem_7.py
import math

def is_prime(num):
    return not (num < 2 or any(num % x == 0 for x in xrange(2, int(num**0.5) + 1)))

current_prime = 0
primes = 0
counter = 0

while(primes < 10001):
    if(is_prime(counter)):
        current_prime = counter
        primes += 1
    counter += 1

print current_prime
